File: src/main/java/pe/codember/app/Challenge02.java
package pe.codember.app;

public class Challenge02 {

  public static final String input = "&###@&*&###@@##@##&######@@#####@#@#@#@##@@@@@@@@@@@@@@@*&&@@@@@@@@@####@@@@@@@@@#########&#&##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@&";

  /*
  "#" Increases the numeric value by 1.
  "@" Decreases the numeric value by 1.
  "*" Multiplies the numeric value by itself.
  "&" Prints the current numeric value.
  */
  public static void main(String[] args) {
    System.out.println(something(input));
  }

  public static String something(String input) {
    int initialValue = 0;
    StringBuilder response = new StringBuilder();

    for (Character letter: input.toCharArray()) {
      switch (letter.toString()) {
        case "#" -> initialValue += 1;
        case "@" -> initialValue -= 1;
        case "*" -> initialValue *= initialValue;
        default -> response.append(initialValue);
      }
    }

    return response.toString();
  }

}

File: src/main/typescript/challenge02.ts
/*
"#" Increases the numeric value by 1.
"@" Decreases the numeric value by 1.
"*" Multiplies the numeric value by itself.
"&" Prints the current numeric value.
*/

const keywords: Record<string, (value: number) => number> = {
  '#': (value: number) => {
    value += 1
    return value
  },
  '@': (value: number) => {
    value -= 1
    return value
  },
  '*': (value: number) => {
    value *= value
    return value
  }
}

export function compileThisWord (input: string): string {
  let result = ''
  let initValue = 0

  for (const letter of input) {
    if (keywords[letter] == null) {
      result += initValue
      continue
    }

    initValue = keywords[letter](initValue)
  }

  return result
}

// switch (letter) {
//   case '#': initValue += 1
//     break
//   case '@': initValue -= 1
//     break
//   case '*': initValue *= initValue
//     break
//   default: result += initValue
//     break
// }
